pkg/db/db.go

```diff
@@ -793,26 +905,495 @@ func (h *DBHandler) RunCustomMigrationAllTables(ctx context.Context, getAllAppsF
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("Error scanning environment locks row from DB. Error: %w\n", err)
+		}
+
+		//exhaustruct:ignore
+		var resultJson = EnvironmentLockMetadata{}
+		err = json.Unmarshal(([]byte)(row.Metadata), &resultJson)
+		if err != nil {
+			return nil, fmt.Errorf("Error during json unmarshal. Error: %w. Data: %s\n", err, row.Metadata)
+		}
+		envLocks = append(envLocks, EnvironmentLock{
+			EslVersion: row.EslVersion,
+			Created:    row.Created,
+			LockID:     row.LockID,
+			Env:        row.Env,
+			Deleted:    row.Deleted,
+			Metadata:   resultJson,
+		})
+	}
+	err = rows.Close()
+	if err != nil {
+		return nil, fmt.Errorf("environment locks: row closing error: %v\n", err)
+	}
+	err = rows.Err()
+	if err != nil {
+		return nil, fmt.Errorf("environment locks: row has error: %v\n", err)
+	}
+	return envLocks, nil // no rows, but also no error
+}
+
+func (h *DBHandler) DBWriteAllEnvironmentLocks(ctx context.Context, transaction *sql.Tx, previousVersion int64, environment string, lockIds []string) error {
+	span, _ := tracer.StartSpanFromContext(ctx, "DBWriteAllEnvironmentLocks")
+	defer span.Finish()
+	slices.Sort(lockIds) // we don't really *need* the sorting, it's just for convenience
+	jsonToInsert, err := json.Marshal(AllEnvLocksJson{
+		EnvLocks: lockIds,
+	})
+	if err != nil {
+		return fmt.Errorf("could not marshal json data: %w", err)
+	}
+	insertQuery := h.AdaptQuery("INSERT INTO all_env_locks (version , created, environment, json)  VALUES (?, ?, ?, ?);")
+	span.SetTag("query", insertQuery)
+	_, err = transaction.Exec(
+		insertQuery,
+		previousVersion+1,
+		time.Now(),
+		environment,
+		jsonToInsert)
+	if err != nil {
+		return fmt.Errorf("could not insert all envs into DB. Error: %w\n", err)
+	}
+	return nil
+}
+
+func (h *DBHandler) DBDeleteEnvironmentLock(ctx context.Context, tx *sql.Tx, environment, lockID string) error {
+	if h == nil {
+		return nil
+	}
+	if tx == nil {
+		return fmt.Errorf("DBDeleteEnvironmentLock: no transaction provided")
+	}
+	span, _ := tracer.StartSpanFromContext(ctx, "DBDeleteEnvironmentLock")
+	defer span.Finish()
+	var previousVersion EslId
+
+	//See if there is an existing lock with the same lock id in this environment. If it exists, just add a +1 to the eslversion
+	existingEnvLock, err := h.DBSelectEnvironmentLock(ctx, tx, environment, lockID)
+
+	if err != nil {
+		return fmt.Errorf("Could not obtain existing environment lock: %w\n", err)
+	}
+
+	if existingEnvLock == nil {
+		previousVersion = 0
+	} else {
+		previousVersion = existingEnvLock.EslVersion
+	}
+
+	//WE simply want to increment the esl version and set the deleted field to 1
+	insertQuery := h.AdaptQuery(
+		"INSERT INTO environment_locks (eslVersion, created, lockID, envName, deleted, metadata) SELECT ?, ?, lockID, envName, ?, metadata FROM environment_locks WHERE eslVersion=? AND envName=? AND lockID=?;")
```

Comment:
I think this can be simplified.
We already obtained the last lock in `existingEnvLock`.
This should give us everything we need.
So this statement could then be a simple `INSERT` without the `SELECT`.